lib/lexer.ts

```diff
@@ -73,12 +73,16 @@ export class Lexer {
     },
 
   ];
+  tokens: OLexerToken[];
+  // A tokens array as a target can be supplied.
   constructor(
     public text: string,
-    public file: OFile
-  ) { }
+    public file: OFile,
+    tokens: [] | undefined = []
```

Comment:
```suggestion
    tokens: OLexerToken[] = []
```

Reply:
Similiar problem hear. I could remove the [] | undefined. But I think it is more clear like this. Added a comment.